src/library/pagination/pagination.component.js

```diff
@@ -0,0 +1,170 @@
+// Import Template
+import template from './pagination.html';
+
+class controller {
+  constructor($log, $scope) {
+    Object.assign(this, {
+      $log,
+      $scope
+    });
+
+    this.currentPage = 0; // binding
+    this.disabled = false; // binding
+    this.itemsPerPage = 0; // binding
+    this.size = ''; // binding
+    this.totalItems = 0; // binding
+    this.visiblePageBuffer = 0; // binding
```

Comment:
mmm... interesting point! never thought about it like that. I like independence...

I'm not sure what do, feel free to leave it in for now, we can always plunk it out later.